templates/components/form/item_itilobject_item_list.html.twig

```diff
@@ -0,0 +1,100 @@
+{#
+ # ---------------------------------------------------------------------
+ #
+ # GLPI - Gestionnaire Libre de Parc Informatique
+ #
+ # http://glpi-project.org
+ #
+ # @copyright 2015-2024 Teclib' and contributors.
+ # @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ #
+ # ---------------------------------------------------------------------
+ #
+ # LICENSE
+ #
+ # This file is part of GLPI.
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # ---------------------------------------------------------------------
+ #}
+
+{% if linknewitil %}
+    <div class='firstbloc'>
+        {{ call('Html::showSimpleForm', [call(itemtype_1 ~ '::getFormURL'), '_add_fromitem', __('New %s for this item')|format(call(itemtype_1 ~ '::getTypeName', [0])), {'itemtype': item.getType(), 'items_id': item.getID()}]) }}
+    </div>
+{% endif %}
+<div>
+{% if number > 0 %}
+    <table class='tab_cadre_fixehov'>
+        {% if readall %}
+            {{ call('Session::initNavigateListItems', [itemtype_1, __(item.getTypeName(1)) ~ " " ~ item.getName()]) }}
+            <tr class='noHover'>
+                <th colspan={{ colspan }}>
+                    {{ __('%d linked %s')|format(number, call(itemtype_1 ~ '::getTypeName', [number])) }}
+                    (<a href="{{ call(itemtype_1 ~ '::getSearchURL') }}?{{ params|url_encode('&') }}">{{ __('Show all') }}</a>)
+                </th>
+            </tr>
+        {% else %}
+            <tr>
+                <th colspan={{ colspan }}>
+                    {{ call(itemtype_1 ~ '::getTypeName', [1]) }} : {{ __("You don't have right to see all") }}
+                </th>
+            </tr>
+        {% endif %}
+{% else %}
+    <table class='tab_cadre_fixehov'>
+        <tr class='noHover'>
+            <th colspan={{ colspan }}>
+                {{ __('No %s on this item')|format(call(itemtype_1 ~ '::getTypeName', [number])) }}
+            </th>
+        </tr>
+{% endif %}
+
+{% if number > 0 %}
+    {{ call(itemtype_1 ~ '::commonListHeader', [html_output])|raw }}
+    {% for data in iterator %}
+        {{ call('Session::addToNavigateListItems', [itemtype_1, data["id"]])|raw }}
+        {{ call(itemtype_1 ~ '::showShort', [data["id"]])|raw }}
+    {% endfor %}
+    {{ call(itemtype_1 ~ '::commonListHeader', [html_output])|raw }}
```

Comment:
I did not try to see how it can be done differently, but if you have to call so many PHP methods to render the page, there is no real value to use Twig for this.

Reply:
Why not use the datatable template? ITILObject class has `getCommonDatatableColumns()` which is the alternative to `commonListHeader` which returns the column and formatter arrays needed for the datatable.

Reply:
Also, the other converted sublists (or at least most of them) do not handle `addToNavigateListItems` calls at all. The feature could be useful to some, but its implementation seems very error-prone because of how it is stored directly in the session. Navigating multiple lists of the same itemtype in multiple tabs and lists of mixed itemtypes don't seem supported at all.